compiler/luci/pass/src/FuseMulWithConvPass.cpp

```diff
@@ -0,0 +1,207 @@
+  assert(indices.size() == 4);
+
+  return indices[0] * node->dim(1).value() * node->dim(2).value() * node->dim(3).value() +
+         indices[1] * node->dim(2).value() * node->dim(3).value() +
+         indices[2] * node->dim(3).value() + indices[3];
+}
+
+/**
+ *  Fuse Mul with Conv if possible
+ *
+ *  NOTE: In case mul is channewise constant, we can try to merge mul with nconv,
+ *
+ *  BEFORE
+ *                    |
+ *              [CircleConv2D] (no activation)
+ *                    |
+ *                  [Mul] (channel-wise/scalar constant)
+ *                    |
+ *
+ *  AFTER
+ *                    |
+ *            [CircleConv2D] (with updated kernels, bias, and activation)
+ *                    |
+ *
+ */
+
+bool fuse_mul_with_conv(luci::CircleMul *mul)
+{
+  // sanity check
+  RETURN_FALSE_UNLESS(mul->dtype() == loco::DataType::FLOAT32);
+
+  luci::CircleConst *const_mul_operand = nullptr;
+  luci::CircleConv2D *conv = nullptr;
+  RETURN_FALSE_UNLESS(luci::fill(&const_mul_operand, &conv).with_commutative_args_of(mul));
+
+  // sanity check
+  RETURN_FALSE_UNLESS(conv->dtype() == loco::DataType::FLOAT32 &&
+                      const_mul_operand->dtype() == loco::DataType::FLOAT32);
+
+  //  NOTE for general activation function: S * Act(A * B) != Act(A*(SB))
+  if (conv->fusedActivationFunction() != luci::FusedActFunc::NONE)
+    return false;
+
+  // check that const_mul_operand is channel-wise constant or just a scalar
+  RETURN_FALSE_UNLESS(const_mul_operand->rank() == 4 || const_mul_operand->rank() == 1 ||
+                      const_mul_operand->rank() == 0);
+
+  std::vector<float> mul_values;
+  if (const_mul_operand->rank() == 4)
+  {
+    // check channel-wise broadcasting
+    RETURN_FALSE_UNLESS(const_mul_operand->dim(0).value() == 1 &&
+                        const_mul_operand->dim(1).value() == 1 &&
+                        const_mul_operand->dim(2).value() == 1);
+  }
+  else if (const_mul_operand->rank() == 1 || const_mul_operand->rank() == 0)
+  {
+    // sanity check
+    RETURN_FALSE_UNLESS(const_mul_operand->size<loco::DataType::FLOAT32>() != 0);
+  }
+
+  mul_values.resize(const_mul_operand->size<loco::DataType::FLOAT32>());
+  for (uint32_t idx = 0; idx < mul_values.size(); idx++)
+  {
+    mul_values[idx] = const_mul_operand->at<loco::DataType::FLOAT32>(idx);
+  }
+
+  // filter
+  auto const conv_filter = dynamic_cast<luci::CircleConst *>(conv->filter());
+
+  // sanity check
+  RETURN_FALSE_UNLESS(conv_filter != nullptr && conv_filter->rank() == 4 &&
+                      conv_filter->dtype() == loco::DataType::FLOAT32);
+
+  auto const out_channels = conv_filter->dim(0).value();
+
+  // multiplier is either channelwise constant or scalar
+  RETURN_FALSE_UNLESS(out_channels == mul_values.size() || mul_values.size() == 1);
+
+  // bias
+  auto const conv_bias = dynamic_cast<luci::CircleConst *>(conv->bias());
+
+  RETURN_FALSE_UNLESS(conv_bias != nullptr && conv_bias->rank() == 1 &&
```

Comment:
I'm not sure this kind of model is valid but but can you plz check this?
- bias with rank 1
- channel > 1
- there is only one element

Reply:
You mean that `bias value` should be broadcasted, (i.e. its a bias is a scalar while `out_channels > 1`)?

Reply:
You mean that `bias value` should be broadcasted, (i.e. its a bias is a scalar while `out_channels > 1`)?

Reply:
Yup

Reply:
I'll fix it. Thank you.

Reply:
Actually, bias broadcasting is optional for this pass and leave a TODO comment.

Reply:
done

Reply:
or leave some TODO comments for later task.

Reply:
> if possible, can you add some more tests for this case?

I'll try.

Reply:
@seanshpark 
Sorry. I failed to produce such a test. For now `luci-interpreter` throws on bias broadcasting:
https://github.com/Samsung/ONE/blob/5d4029a49efdb1a8442d4dd6ad023713f69d47ca/compiler/luci-interpreter/src/kernels/Conv2D.cpp#L94-L95.

Reply:
@seanshpark 
Should i add comments mentioned in https://github.com/Samsung/ONE/pull/11993#discussion_r1390806220?

Reply:
However gerenerated `circle` model with bias broadcated can be successfully processed by the pass. Pleasee see results attached:
[fuse_mul_conv_with_bias_broadcasted.zip](https://github.com/Samsung/ONE/files/13333389/fuse_mul_conv_with_bias_broadcasted.zip)

Reply:
I think it would be better to follow the luci-interpreter implementation.

Reply:
You mean to check bias as in `luci-interpreter`?

Reply:
> Should i add comments mentioned in https://github.com/Samsung/ONE/pull/11993#discussion_r1390806220?

I think LUCI_INTERPRETER_CHECK condition is enough as it describes possible cases.